instrumentation/netty/netty-4.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/Netty41ClientSslTest.java

```diff
@@ -0,0 +1,260 @@
+                ChannelPipeline pipeline = socketChannel.pipeline();
+
+                SslContext sslContext = SslContextBuilder.forClient().build();
+                SSLEngine sslEngine = sslContext.newEngine(socketChannel.alloc());
+                if (enabledProtocols != null) {
+                  sslEngine.setEnabledProtocols(enabledProtocols.toArray(new String[0]));
+                }
+                pipeline.addLast(new SslHandler(sslEngine));
+
+                pipeline.addLast(new HttpClientCodec());
+              }
+            });
+    return bootstrap;
+  }
+
+  @BeforeEach
+  void setUp() {
+    server = new HttpClientTestServer(testing.getOpenTelemetry());
+    server.start();
+    eventLoopGroup = new NioEventLoopGroup();
+    System.setProperty("otel.instrumentation.netty.connection-telemetry.enabled", "true");
+    System.setProperty("otel.instrumentation.netty.ssl-telemetry.enabled", "true");
+  }
+
+  @AfterEach
+  void tearDown() throws ExecutionException, InterruptedException, TimeoutException {
+    server.stop().get(10, TimeUnit.SECONDS);
+    eventLoopGroup.shutdownGracefully();
+    System.clearProperty("otel.instrumentation.netty.connection-telemetry.enabled");
+    System.clearProperty("otel.instrumentation.netty.ssl-telemetry.enabled");
+  }
+
+  @Test
+  @DisplayName("should fail SSL handshake")
+  @SuppressWarnings("InterruptedExceptionSwallowed")
+  public void testFailSslHandshake() throws Exception {
+    Bootstrap bootstrap = createBootstrap(eventLoopGroup, Collections.singletonList("SSLv3"));
+    URI uri = server.resolveHttpsAddress("/success");
+    DefaultFullHttpRequest request =
+        new DefaultFullHttpRequest(
+            HttpVersion.HTTP_1_1, HttpMethod.GET, uri.getPath(), Unpooled.EMPTY_BUFFER);
+    request.headers().set(HttpHeaderNames.HOST, uri.getHost() + ":" + uri.getPort());
+
+    Throwable thrownException = null;
+    try {
+      testing.runWithSpan(
+          "parent",
+          () -> {
+            Channel channel = bootstrap.connect(uri.getHost(), uri.getPort()).sync().channel();
+            CompletableFuture<Integer> result = new CompletableFuture<>();
+            channel.pipeline().addLast(new ClientHandler(result));
+            channel.writeAndFlush(request).get(10, TimeUnit.SECONDS);
+            result.get(10, TimeUnit.SECONDS);
+          });
+    } catch (Throwable e) {
+      thrownException = e;
+    }
+
+    List<AttributeAssertion> attributeAssertion =
+        asList(
+            equalTo(ExceptionAttributes.EXCEPTION_MESSAGE, thrownException.getCause().getMessage()),
+            satisfies(ExceptionAttributes.EXCEPTION_STACKTRACE, v -> v.isNotNull()),
+            equalTo(
+                ExceptionAttributes.EXCEPTION_TYPE,
+                thrownException.getCause().getClass().getCanonicalName()));
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
```

Comment:
groovy test sorts RESOLVE/CONNECT/SSL handshake spans. Are you sure that you don't need to do this here? Won't this make the test flaky?

Reply:
the test works fine

Reply:
I think the concern is that it might work fine most of the time, but it could also become flaky and result in sporadic failures

Reply:
> I think the concern is that it might work fine most of the time, but it could also become flaky and result in sporadic failures

let's fix it more stable.

Reply:
Sorry, after thinking about this a bit more could you change it back to `hasSpansSatisfyingExactly`. I think the underlying issue that caused the spans to appear in different order was probably resolved so sorting the spans shouldn't be necessary. If it turns out to be necessary we can fix it later.

Reply:
I will revert it back